tests/pso_test.cpp

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file pso_test.cpp
+ * @author Suryoday Basak
+ * @author Chintan Soni
+ *
+ * Test file for PSO optimizer.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <ensmallen.hpp>
+#include "catch.hpp"
+
+using namespace ens;
+using namespace ens::test;
+
+TEST_CASE("LBestPSOSphereFunctionTest", "[PSOTest]")
```

Comment:
This looks really good, would be interesting to see if the current version is able to minimise: ``SchafferFunctionN2`` and  ``SchafferFunctionN4`` as well.

Reply:
Sure it does, but I have noted that the swarm is initialized either over the optimum or close to the optimum. It had kind of slipped my mind, but I need to tweak the API to incorporate user-selected lower-bounds and upper bounds. I don't want to make it too complicated, I'll initialize the population within a hypercube.

Reply:
Once I implement this, I'll deliberately initialize the swarm far from the optimum. If it succeeds then, we will know that it is fairly robust.